### ISSUE TITLE
Stable functions not re-evaluated

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#2661]
If you have a stable function in a `INSERT ... VALUES (<stable function>)` statement, the function is evaluated once, at planning time, instead of in each execution. For example:

```
postgres=# create table inserttesttab (t timestamp);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 't' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE TABLE
postgres=# PREPARE myinsert AS INSERT INTO inserttesttab values (transaction_timestamp());
PREPARE
postgres=# execute myinsert;
INSERT 0 1
postgres=# execute myinsert;
INSERT 0 1
postgres=# execute myinsert;
INSERT 0 1
postgres=# execute myinsert;
INSERT 0 1
postgres=# select * from inserttesttab ;
             t              
----------------------------
 2017-06-21 20:53:50.655385
 2017-06-21 20:53:50.655385
 2017-06-21 20:53:50.655385
 2017-06-21 20:53:45.652178
(4 rows)
```

All those timestamps should've been different. Same issue can be seen with a user-defined function that contains an INSERT like that.

**Comment by asimrp** [comment]
Why are only three time stamps identical in the output and not all four?  Is the function getting evaluated twice?

**Comment by hlinnaka** [comment]
Yes, the function is evaluated twice. The first invocation produces one value, and all subsequent invocation produce the same value. I'm not sure why that happens.

**Comment by ivannovick** [comment]
I think i have heard about this for a long time, and its a bug (opportunity for improvement)

**Comment by hlinnaka** [comment]
Fixed by PR #2670.
